services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { FC, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { change, Field, formValueSelector, InjectedFormProps, reduxForm } from "redux-form";
+import { Alert, Col, Row, Typography } from "antd";
+import { Form } from "@ant-design/compatible";
+import { remove } from "lodash";
+import { ENVIRONMENT, IProject } from "@mds/common";
+import * as API from "@mds/common/constants/API";
+import {
+  createInformationRequirementsTable,
+  updateInformationRequirementsTableByFile,
+} from "@mds/common/redux/actionCreators/projectActionCreator";
+import { getProject } from "@mds/common/redux/selectors/projectSelectors";
+import { MODERN_EXCEL } from "@mds/common/constants/fileTypes";
+import * as FORM from "@/constants/forms";
+import LinkButton from "@/components/common/LinkButton";
+import DocumentTable from "@/components/common/DocumentTable";
+import IRTFileUpload from "@/components/Forms/projects/informationRequirementsTable/IRTFileUpload";
+import {
+  renderCategoryColumn,
+  renderDateColumn,
+  renderTextColumn,
+} from "@mds/common/components/common/CoreTableCommonColumns";
+import { formatDateTime } from "@common/utils/helpers";
+import { documentNameColumn } from "@/components/common/DocumentColumns";
+import { MineDocument } from "@mds/common/models/documents/document";
+
+interface IRTFileImportProps {
+  projectGuid: string;
+  informationRequirementsTableDocumentTypesHash: any;
+  importIsSuccessful: (success: boolean, err: string) => void;
+  downloadIRTTemplate: (url: string) => void;
+}
+
+export const IRTFileImport: FC<IRTFileImportProps & InjectedFormProps<any>> = ({
```

Comment:
All changes here are refactoring from class based component, implementing typescript and refactoring to use react-redux hooks.  All functionality is the same.

Reply:
My heart wants to get rid of old antd form too, but this is a great step forward.

Reply:
Switched it over!